ICU 55 and 57 on Gentoo Linux say: 'ucol_setVariableTop' is deprecated

@ellismg,

I setup Gentoo [LXC container](https://linuxcontainers.org/) on Ubuntu. The step-by-step guide to setup Gentoo and repro this issue (from scratch to start building CoreCLR) is documented in [this gist](https://gist.github.com/jasonwilliams200OK/1a2e2c0e904ffa95faf6333fcd88d9b8) *(updated this guide with additional steps to build CoreFX native)*.

Failure Message:

```sh
[  3%] Building CXX object src/corefx/System.Globalization.Native/CMakeFiles/System.Globalization.Native.dir/localeNumberData.cpp.o
/root/coreclr/src/corefx/System.Globalization.Native/collation.cpp:282:9: error: 
      'ucol_setVariableTop' is deprecated [-Werror,-Wdeprecated-declarations]
        ucol_setVariableTop(pClonedCollator, ignoreSymbolsVariableTop, 1, pErr);
        ^
/usr/include/unicode/ucol.h:1254:1: note: 'ucol_setVariableTop' has been
      explicitly marked deprecated here
ucol_setVariableTop(UCollator *coll, 
^
1 error generated.
src/corefx/System.Globalization.Native/CMakeFiles/System.Globalization.Native.dir/build.make:110: recipe for target 'src/corefx/System.Globalization.Native/CMakeFiles/System.Globalization.Native.dir/collation.cpp.o' failed
make[2]: *** [src/corefx/System.Globalization.Native/CMakeFiles/System.Globalization.Native.dir/collation.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

Official documentation suggests the same thing:
http://icu-project.org/apiref/icu4c/ucol_8h.html#a0e9c28a8b8b88890722daca111207688

> Deprecated:
    ICU 53 Call ucol_setMaxVariable() instead. 

(`ucol_setMaxVariable()` is stable since `ICU 53`)

My understand is the decision was made based on this proposal: https://sourceforge.net/p/icu/mailman/message/29538820/.

@adityamandaleeka, seems like that's the case. `cmake_symbol_exists` detection for `ucol_setMaxVariable` is failing. From CMakeError.log:

>
/root/coreclr/bin/obj/Linux.x64.Debug/CMakeFiles/CMakeTmp/CheckSymbolExists.c:(.text+0x19): undefined reference to `ucol_setMaxVariable'

I tried using `cmake_function_exists` (which uses the built in `-DCHECK_FUNCTION_EXISTS`), but got the exact same error. It seems like it is not linking to libicu (even though I had `
set(CMAKE_REQUIRED_LIBRARIES ${ICUUC})` before it):

> /usr/bin/clang   -Wall -std=c11 -DCHECK_FUNCTION_EXISTS=ucol_setMaxVariable -I/usr/local/opt/icu4c/include    -o CMakeFiles/cmTC_da2df.dir/CheckFunctionExists.c.o   -c /usr/share/cmake/Modules/CheckFunctionExists.c

--- a/src/corefx/System.Globalization.Native/configure.cmake
+++ b/src/corefx/System.Globalization.Native/configure.cmake
@@ -8,11 +8,18 @@ CHECK_CXX_SOURCE_COMPILES("
     int main() { UDateFormatSymbolType e = UDAT_STANDALONE_SHORTER_WEEKDAYS; }
 " HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS)
 
+if(NOT CLR_CMAKE_PLATFORM_DARWIN)
+    set(CMAKE_REQUIRED_LIBRARIES ${ICUUC} ${ICUI18N})
+else()
+    set(CMAKE_REQUIRED_LIBRARIES ${ICUCORE})
+endif()
+
 check_symbol_exists(
     ucol_setMaxVariable
     "unicode/ucol.h"
     HAVE_SET_MAX_VARIABLE)
 
+unset(CMAKE_REQUIRED_LIBRARIES)
 unset(CMAKE_REQUIRED_INCLUDES)
 
 configure_file(

